### PR TITLE
fix: fix protocol to adapter file path conversion

### PR DESCRIPTION
**Pull request by louise-poole**
and update docs and remove dead code clippy skips.

### CODE REVIEW

**Review by dianacarvalho1 — COMMENTED** [review]
Thank you 🙏🏼

File: Readme.md [hunk 1]
```diff
@@ -13,26 +13,31 @@ protocols, but it is not necessarily limited to this.
...
 
 ## Adding a new Protocol
 
 To add a new protocol, you will need to complete the following high-level steps:
 
+#### Native protocol:
+
 1. Create a protocol state struct that implements the `ProtocolSim` struct.
-2. _(legacy information)_ Add associated events and implement the transition method on the protocol state struct.
+2. Create a tycho decoder for the protocol state: i.e. implement `TryFrom` for `ComponentWithState` to your new
```
Comment (kayibal) [inline Readme.md:31]: Is it `TryFrom` or `TryFromWithBlock`, also we should specify the error type of the implementation here I think.

File: src/protocol/vm/tycho_decoder.rs [hunk 2]
```diff
@@ -113,15 +117,15 @@ impl TryFromWithBlock<ComponentWithState> for VMPoolState<PreCachedDB> {
...
-                    .protocol_system
-                    .as_str(),
-            )
-        );
+        let protocol_name = snapshot
+            .component
+            .protocol_system
+            .strip_prefix("vm:")
+            .unwrap_or({
+                Err(InvalidSnapshotError::VMError(SimulationError::DecodingError("Invalid protocol system name: VMPoolStates expect a protocol name starting with 'vm:'".to_string())))?
```
Comment (kayibal) [inline src/protocol/vm/tycho_decoder.rs]: What err inception is going on here? Did you just look for the closest DecodingError?

File: src/protocol/vm/tycho_decoder.rs [hunk 3]
```diff
@@ -113,15 +117,15 @@ impl TryFromWithBlock<ComponentWithState> for VMPoolState<PreCachedDB> {
...
-                snapshot
-                    .component
-                    .protocol_system
-                    .as_str(),
-            )
-        );
+        let protocol_name = snapshot
+            .component
+            .protocol_system
+            .strip_prefix("vm:")
```
Comment (kayibal) [inline src/protocol/vm/tycho_decoder.rs:123]: What if we have a VM and a native implementation?

File: Readme.md (hunk 1 above)
Comment (louise-poole) [reply Readme.md:31]: It's `TryFrom`. `TryFromWithBlock is only used by VmPoolState which already has a decoder.

File: src/protocol/vm/tycho_decoder.rs (hunk 2 above)
Comment (louise-poole) [reply src/protocol/vm/tycho_decoder.rs]: `InvalidSnapshotError` only has 'MissingAttribute', 'VMError' and 'ValueError'. Initially felt VMError was the most appropriate, but a simple ValueError is probably better?

File: src/protocol/vm/tycho_decoder.rs (hunk 3 above)
Comment (louise-poole) [reply src/protocol/vm/tycho_decoder.rs:123]: True! I'll make the stripping optional.

Comment (louise-poole) [reply src/protocol/vm/tycho_decoder.rs:123]: On that note, is there a reason we still need the 'vm:' prefix for these protocol names in tycho?

**Review by dianacarvalho1 — APPROVED** [review]

File: Readme.md (hunk 1 above)
Comment (kayibal) [reply Readme.md:31]: Would prefer if we were consistent

File: src/protocol/vm/tycho_decoder.rs (hunk 3 above)
Comment (kayibal) [reply src/protocol/vm/tycho_decoder.rs:119]: I guess if we have both a VM and native implementation we could distinguish them like this.

File: Readme.md (hunk 1 above)
Comment (louise-poole) [reply Readme.md:31]: Okay, I'll do this in a separate PR... Diana is waiting on this one.

**Pull request merged**